src/coreclr/System.Private.CoreLib/src/System/Enum.CoreCLR.cs

```diff
@@ -20,7 +21,54 @@ public abstract partial class Enum
         private extern CorElementType InternalGetCorElementType();
 
         [MethodImpl(MethodImplOptions.InternalCall)]
-        internal static extern RuntimeType InternalGetUnderlyingType(RuntimeType enumType);
+        internal static extern RuntimeType InternalGetUnderlyingTypeImpl(RuntimeType enumType);
+
+        internal static RuntimeType InternalGetUnderlyingType(RuntimeType enumType)
```

Comment:
What is the performance impact of this change on `Type.GetEnumUnderlyingType`?

If you are after optimizing `Type.GetEnumUnderlyingType` performance, I think it is better to do it by going via CorElementType. CorElementType is native to the runtime and cheap to get. GetTypeCode is more expensive to get.

Reply:
Was going to make benchmarks after checking if everything passes on the CI, This should be free when `GetTypeCode` gets handled as an intrinsic after inlining, and even without it since it's cached a call to it was around 6ns when benchmarked on .Net 6.

Reply:
The typical case is that this is called on a type that cannot be statically determined. None of the intrinsic expansions kick in for those cases.

Reply:
I'd assume that it can also happen in generic contexts, when the JIT will know the exact type.

Reply:
@jkotas I've removed changes conflicting with your PR #71873, now this only affects constant typeofs.